### ISSUE TITLE
Unable to resolve  java qlpacks after v2.15.2

### ISSUE THREAD

**Issue by chmodxxx** [github/codeql#14788]
Hello,
 
We have a customizations chain depending on java-all and we are now facing some errors for qlpacks resolution after upgrading to 2.15.2 (for reference, this wasn't an issue in 2.15.1) . 
The error thrown is : 

```
> Task :preCompilePack FAILED
ERROR: Referenced pack 'codeql/rangeanalysis' was not found locally, and no lock file is present for 'org/some-ql-java'. Run 'codeql pack install' to create the lock file and download the dependencies. (/home/foo/some-ql-java/src/qlpack.yml:1,1-1)
ERROR: No valid pack solution found:
Because 'codeql/java-queries' depends on 'codeql/java-all@*', which depends on 'codeql/rangeanalysis@0.0.1', 'codeql/java-queries' requires 'codeql/rangeanalysis@0.0.1'.
So, because 'codeql/rangeanalysis' was not found. and 'org/some-ql-java' depends on 'codeql/java-queries@*', version solving failed.
 (/home/foo/some-ql-java/src/qlpack.yml:1,1-1)
A fatal error occurred: Could not resolve library path for /home/foo/some-ql-java/src
```
Is there a breaking change impacting qlpack resolution i'm not aware of ? I read the changelog but nothing relevant to java-all or java-queries

**Comment by chmodxxx** [comment]
Ah I see, so now we have  codeql/rangeanalysis as dependencies in java-all , the issue is that it is not in the qlpacks folder. The only way I see that would solve this is to run codeql qlpack install, but this is not ideal for our workflows since we typically don't want to run arbitrary installs over our CI/CD build pipeline, it would be better if the rangeanalysis pack was also in the qlpack folder. 

Let me know if that is possible to fix

**Comment by mbg** [comment]
Hi @chmodxxx 👋 

Just to check, have you run `codeql pack install` in the folder containing your custom `qlpack.yml` as suggested by the error? If so, what's the output of that and what's the resulting `codeql-pack.lock.yml`?

**Comment by chmodxxx** [comment]
yeah, the issue is that that might solve things locally, but we run builds to generate our customized packs in restricted environments that might not be able to reach the internet, so typically we would expect the rangeanalysis pack to be present in the folders qlpacks so that the resolution can work  locally (from files without an additional install step) . 
Does that make sense ?

**Comment by mbg** [comment]
Hi @chmodxxx, yes that makes sense and apologies. I only saw your other reply after I had already submitted my response to your original post!

The CodeQL bundle for 2.15.2 at https://github.com/github/codeql-action/releases/download/codeql-bundle-v2.15.2/codeql-bundle.tar.gz does include the required `rangeanalysis` pack. Is that what you are using? If not, which `qlpacks` directory are you looking at that you aren't finding `./codeql/qlpacks/codeql/rangeanalysis/0.0.1/qlpack.yml` in?
